apps/src/craft/code-connection/craft.js

```diff
@@ -119,13 +120,30 @@ Craft.runButtonClick = function () {
 Craft.executeUserCode = function () {
   let codeBlocks = Blockly.mainBlockSpace.getTopBlocks(true);
   const code = Blockly.Generator.blocksToCode('JavaScript', codeBlocks);
+  let interpreter;
 
   const evalApiMethods = {
     log: function (blockID, value) {
       studioApp().highlight(blockID);
       console.log('Logged: ' + value);
     },
+    prompt: function (blockID, callback) {
+      studioApp().highlight(blockID);
+      const value = prompt('Enter a value:');
+      setTimeout(() => {
+        callback(value);
+        interpreter.run();
+      }, 0);
+    },
+    delay: function (blockID, milliseconds, callback) {
+      studioApp().highlight(blockID);
+      setTimeout(() => {
+        callback();
+        interpreter.run();
+      }, milliseconds);
+    }
   };
 
-  codegen.evalWith(code, evalApiMethods);
+  codegen.asyncFunctionList = [evalApiMethods.prompt, evalApiMethods.delay];
```

Comment:
I just got rid of `asyncFunctionList` since it wasn't being used anywhere. If you need to add that functionality back, add tests along with it.